Tools/CMake/R.cmake

```diff
@@ -0,0 +1,743 @@
+# R.cmake handles the process of downloading, patching, locating and pathing the R
+# instance in different platforms. There is a lot that is going on here, so, if you
+# don't know what you are doing, you might very well start breaking things!
+#
+# The general flow of setting R is as follow:
+#
+#   - Downloading or locating the R instance, e.g., R.framework,
+#   - Copying the R instance to the build folder, after patching and preparing it (only on Windows and macOS)
+#   - Installing RInside, and Rcpp
+#   - Interpolating all the necessary paths and passing them to the rest of the CMake
+#
+# On macOS,
+#
+#
+# On Windows,
+#
+#
+# On Linux,
+#
+# Todos:
+#
+# - [ ] Maybe, the entire R.framework prepration should be a target. The advantages
```

Comment:
I'm not still sure about this, there are advantages for it to be prepared at the config time, and it fits to the description of a config operation better than the build operation. CMake really like things very CMake-y, and deviating from it can only cause headache, that's why I didn't do it, but made a note of it.

Reply:
Well it certainly works now so I dont mind